packages/blue-sdk/src/position/PreLiquidatablePosition.ts

```diff
@@ -0,0 +1,322 @@
+import type { Address } from "viem";
+import { ORACLE_PRICE_SCALE } from "../constants";
+import {
+  CapacityLimitReason,
+  type Market,
+  MarketUtils,
+  type MaxBorrowOptions,
+  type MaxWithdrawCollateralOptions,
+} from "../market";
+import { MathLib, SharesMath } from "../math";
+import { AccrualPosition, type IAccrualPosition } from "./Position";
+
+export type PreLiquidationParams = {
+  preLltv: bigint;
+  preLCF1: bigint;
+  preLCF2: bigint;
+  preLIF1: bigint;
+  preLIF2: bigint;
+  preLiquidationOracle: Address;
+};
+
+export class PreLiquidatablePosition extends AccrualPosition {
+  /**
+   * The pre-liquidation parameters.
+   */
+  public readonly preLiquidationParams: PreLiquidationParams;
+
+  /**
+   * The address of the preLiquidation contract.
+   */
+  public readonly preLiquidation: Address;
+
+  constructor(
+    position: IAccrualPosition,
+    market: Market,
+    preLiquidationParams: PreLiquidationParams,
+    preLiquidation: Address,
+  ) {
+    super(position, market);
+
+    this.preLiquidationParams = preLiquidationParams;
+    this.preLiquidation = preLiquidation;
+  }
+
+  /**
+   * Whether this position is healthy.
+   * `undefined` iff the market's oracle is undefined or reverts.
+   */
+  get isLiquidatable() {
+    return this.market.isHealthy(this);
+  }
+
+  /**
+   * Whether this position is preLiquidatable.
+   * `undefined` iff the market's oracle is undefined or reverts.
+   */
+  get isPreLiquidatable() {
+    if (this.ltv === undefined || this.ltv === null) return undefined;
+
+    return (
+      this.ltv > this.preLiquidationParams.preLltv &&
+      this.ltv < this.market.params.lltv
```

Comment:
```suggestion
      this.ltv <= this.market.params.lltv
```
 Shouldn't it be lte?